src/main/ruby/core/process_mirror.rb

```diff
@@ -430,10 +430,43 @@ def spawn
               end
             end
             raise Errno::ENOENT, "No such file or directory - #{@command}"
+          elsif pid < 0
+            raise SystemCallError.new("posix_spawnp #{@command}", -pid)
```

Comment:
This dual use of PID and inverting the sign of the return value is a bit strange to look at. Is there a way to get another outval for the return value separate from the PID?

Reply:
As an aside, I actually think all syscalls should follow this approach (and they often do at the kernel level but the libc returns `-1` + errno instead for most syscalls).
We could pass an extra pointer argument, it just seems more complex to deal with.
Negative PIDs are always invalid anyway.
Note that `posix_spawn` is also a bit different compared to other native calls because it returns the `errno number` rather than setting errno.
I prefer returning an invalid value than reading errno, as reading errno is more fragile (it could be set by any call between `posix_spawn` and the end of `truffleposix_posix_spawnp`).